File: camerax_android/android/src/main/kotlin/dev/hebei/camerax_android/core/ImageFormat.kt
package dev.hebei.camerax_android.core

enum class ImageFormat {
    JPEG, JPEG_R, YUV_420_888, RGBA_8888,
}

val Int.imageFormatArgs
    get() = when (this) {
        android.graphics.ImageFormat.JPEG -> ImageFormat.JPEG
        android.graphics.ImageFormat.JPEG_R -> ImageFormat.JPEG_R
        android.graphics.ImageFormat.YUV_420_888 -> ImageFormat.YUV_420_888
        android.graphics.ImageFormat.FLEX_RGBA_8888 -> ImageFormat.RGBA_8888
        else -> throw IllegalArgumentException()
    }

File: camerax_android/lib/src/core/image_format.dart
// import 'package:camerax_android/src/jni.dart' as $native;
// import 'package:camerax_platform_interface/camerax_platform_interface.dart'
//     as $interface;

// extension ImageFormatArgs on $interface.ImageFormat {
//   $native.ImageFormat get obj {
//     switch (this) {
//       case $interface.ImageFormat.jpeg:
//         return $native.ImageFormat.JPEG!;
//       case $interface.ImageFormat.jpegR:
//         return $native.ImageFormat.JPEG_R!;
//       case $interface.ImageFormat.yuv420_888:
//         return $native.ImageFormat.YUV_420_888!;
//       case $interface.ImageFormat.rgba8888:
//         return $native.ImageFormat.RGBA_8888!;
//     }
//   }
// }

// extension ImageFormatObj on $native.ImageFormat {
//   $interface.ImageFormat get args {
//     if (this == $native.ImageFormat.JPEG) {
//       return $interface.ImageFormat.jpeg;
//     }
//     if (this == $native.ImageFormat.JPEG_R) {
//       return $interface.ImageFormat.jpegR;
//     }
//     if (this == $native.ImageFormat.YUV_420_888) {
//       return $interface.ImageFormat.yuv420_888;
//     }
//     if (this == $native.ImageFormat.RGBA_8888) {
//       return $interface.ImageFormat.rgba8888;
//     }
//     throw ArgumentError.value(this);
//   }
// }
